pkg/controller/operators/operator_controller_test.go

```diff
@@ -77,16 +75,62 @@ var _ = Describe("Operator Controller", func() {
 			})
 
 			It("should re-create it", func() {
-				Eventually(func() error {
-					return k8sClient.Get(ctx, name, operator)
-				}).Should(Succeed())
+				// There's a race condition between this test and the controller. By the time,
+				// this function is running, we may be in one of three states.
+				//   1. The original deletion in the test setup has not yet finished, so the original
+				//      operator resource still exists.
+				//   2. The operator doesn't exist, and the controller has not yet re-created it.
+				//   3. The operator has already been deleted and re-created.
+				//
+				// This race condition means we need to test for either of two possibilities when getting
+				// the operator:
+				//   a. We see a "not found" error followed by a nil error. This covers 1) and 2) above.
+				//   b. We eventually timeout looking for a), always seeing a nil error. This covers 3) above.
+
+				// This section waits until we've seen a not found error followed by a nil error. It
+				// returns any other unexpected error immediately.
+				gotNotFound := false
+				err := wait.PollImmediate(interval, timeout, func() (bool, error) {
```

Comment:
Could we use [Gomega's Eventually](https://onsi.github.io/gomega/#eventually) here instead of `wait.PollImmediate`?

Reply:
I tried several different ways of making `Eventually` work, but I was always foiled by the fact that this test basically is `Eventually(notFoundThenFound) OR Consistently(found)`.

There doesn't seem to be a way with Gomega to compose Assertions in this way, but I'd be happy to be proven wrong because that would make this test more clear.